docs/content/guide/component.ngdoc

```diff
@@ -0,0 +1,453 @@
+      }
+    ```
+  - That way, the parent component can decide what to do with the event (e.g. delete an item or update the properties)
+    ```js
+      ctrl.deleteHero(hero) {
+        $http.delete(...).then(function() {
+          var idx = ctrl.list.indexOf(hero);
+          if (idx >= 0) {
+            ctrl.list.splice(idx, 1);
+          }
+        });
+      }
+      ```
+
+- An application is a tree of components
+Ideally, the whole application should be a tree of components. If you minimize two-way data binding,
+and implement an uni-directional data flow, it's easier to predict when data changes and what the state
+of a component is.
+
+## Example of a component tree
+
+The following example expands on the simple component example and incorporates the concepts we introduced
+above:
+
+In addition to the heroDetail component, we now have a heroList component that holds the data of
+different heros, and creates a heroDetail for each of them.
+
+The heroDetail component now contains the following new functions:
+- a delete button that directly calls the bound onDelete function of the heroList component
+- an input to change the her location, in the form of a reusable editableField component
+
+<example name="heroComponentTree" module="heroApp">
+<file name="index.js">
+  var mode = angular.module('heroApp', []);
+</file>
+
+<file name="heroList.js">
+  var myMod = angular.module('heroApp');
+
+  function HeroListController($scope, $element, $attrs) {
+    var ctrl = this;
+
+    // This would be loaded by $http etc.
+    ctrl.list = [
+      {
+        name: 'Superman',
+        location: 'Fortress of solitude'
+      },
+      {
+        name: 'Batman',
+        location: 'Wayne Manor'
+      }
+    ];
+
+    ctrl.heroExists = function() {
+      var idx = ctrl.list.indexOf(hero);
+      return idx >= 0;
+    };
+
+    ctrl.updateHero = function(hero, prop, value) {
+      var idx = ctrl.list.indexOf(hero);
+      if (idx >= 0) {
+        ctrl.list[idx][prop] = value;
+      }
+    };
+
+    ctrl.deleteHero = function(hero) {
+      var idx = ctrl.list.indexOf(hero);
+      if (idx >= 0) {
+        ctrl.list.splice(idx, 1);
+      }
+    };
+  }
+
+  myMod.component('heroList', {
+    templateUrl: 'heroList.html',
+    controller: HeroListController
+  });
+
+</file>
+
+<file name="heroDetail.js">
+  var myMod = angular.module('heroApp');
+
+  function HeroDetailController($scope, $element, $attrs) {
+    var ctrl = this;
+
+    this.$onInit = function() {
+      if (!this.hero.location) {
+        this.hero.location = 'unknown';
+      }
+    };
+  }
+
+  myMod.component('heroDetail', {
+    templateUrl: 'heroDetail.html',
+    controller: HeroDetailController,
+    bindings: {
+      hero: '=',
+      onDelete: '&',
+      onUpdate: '&'
+    }
+  });
+</file>
+
+<file name="editableField.js">
+
+   function EditableFieldController($scope, $element, $attrs) {
+    var ctrl = this;
```

Comment:
I prefer `ctrl` (in fact I prefer `self`, but `ctrl` is preferrable to `$ctrl`).
It's shorter/easier to type and doesn't give the false impression that the view alias and the local variable names must match.